src/API.jl

```diff
@@ -894,71 +894,88 @@ end
 
 precompile() = precompile(Context())
 function precompile(ctx::Context)
+    
+    function is_stale(paths, sourcepath)
+        for path_to_try in paths::Vector{String}
+            staledeps = Base.stale_cachefile(sourcepath, path_to_try, Base.TOMLCache())
```

Comment:
Perhaps hoist the creation of the TOMLCache?
```suggestion
        toml_c = Base.TOMLCache()
        for path_to_try in paths::Vector{String}
            staledeps = Base.stale_cachefile(sourcepath, path_to_try, toml_c)
```

Reply:
Could it be hoisted even higher? Generated once at the start of the precompile job?

Reply:
Maybe? I wasn't sure about that since you spawn multiple tasks.

Reply:
True. I didn't realize it was an empty object for preallocation. I thought it was a snapshot of some form of system state for reference. I guess I don't know enough to answer your original question/suggestion

Reply:
The suggestion in the diff is safe, but I don't think you can move it out further since then multiple tasks will use the same object.